certs/certs.go

```diff
@@ -41,12 +41,9 @@ const (
 // locally authorizing clients.  If 'server' is true, it writes out certs
 // which can be used to verify the server, otherwise it writes out certs
 // clients can use to authorize themselves to the server.
-func WriteNewCerts(certFile, keyFile string, server bool) error {
-	// Implementation mostly taken from http://golang.org/src/pkg/crypto/tls/generate_cert.go
-	priv, err := rsa.GenerateKey(rand.Reader, bits)
-	if err != nil {
-		return fmt.Errorf("failed to generate private key: %v", err)
-	}
+func WriteNewCerts(certFile, keyFile string, caCertFile string, server bool, is_ca bool ) error {
```

Comment:
Make sure to run `go fmt` on any Go code.  Note that unnecessary space after `is_ca bool`.  Also, a few Go style comments:

*   If you've got two args, skip the type on the first one:  `server, is_ca bool` instead of `server bool, is_ca bool`
*   Use CamelCase for args:  `isCA` instead of `is_ca`

For server/client/CA, I'd use an enum instead of a set of bools:

    type CertType int
    const (
      CA CertType = iota
      Server
      Client
    )